etl/chart_revision/v2/updaters/variable_update.py

```diff
@@ -465,57 +313,83 @@ def find_charts_from_variable_ids(variable_ids: Set[int]) -> List[gm.Chart]:
     return charts
 
 
-def _get_time_ranges(
-    config: Dict[str, Any], variable_mapping: Dict[int, int], variable_meta: Dict[int, Any], properties: List[str]
-) -> List[Tuple[int, int]]:
-    """Get time ranges of variables used in the chart configuration.
+def update_chart_config(
+    config: Dict[str, Any],
+    indicator_mapping: Dict[int, int],
+    schema: Optional[Dict[str, Any]] = None,
+) -> Dict[str, Any]:
+    """Update indicator references to the new ones.
 
-    It obtains the list of ranges with the currently in use variables and the future set of variables in use.
-    """
-    # ranges_old = []
-    ranges_new = []
-    # Iterate over all dimensions used in the chart, obtain their year ranges, and add to range list.
-    # Also check if variable will be updated, so that the new range is added to ranges_new.
-    for dim in config["dimensions"]:
-        if dim["property"] in properties:
-            variable_id = dim["variableId"]
-            # ranges_old.append([variable_meta[variable_id]["minYear"], variable_meta[variable_id]["maxYear"]])
-            if variable_id in variable_mapping:
-                variable_id_new = variable_mapping[variable_id]
-                ranges_new.append(
-                    (
-                        variable_meta[variable_id_new]["minYear"],
-                        variable_meta[variable_id_new]["maxYear"],
-                    )
-                )
-    return ranges_new
-
-
-def intersect_range(ranges: List[Tuple[int, int]]) -> Tuple[int, int]:
-    """Get the range that is common to all ranges in `ranges`.
-
-    It performs INTERSECT operation on the ranges.
+    The chart config contains some fields that point to the indicators in use. In the attempt to migrating these, we should update all references to the new indicators.
     """
-    range_min = None
-    range_max = None
-    for range_ in ranges:
-        if range_min is None or range_[0] > range_min:
-            range_min = range_[0]
-        if range_max is None or range_[1] < range_max:
-            range_max = range_[1]
-    return cast(int, range_min), cast(int, range_max)
-
-
-def union_range(ranges: List[Tuple[int, int]]) -> Tuple[int, int]:
-    """Get the range that contains all ranges in `ranges`.
-
-    It performs UNION operation on the ranges.
+    # Fix errors in schema
+    config_new = fix_errors_in_schema(config)
+    # Validate config agains schema
+    if schema is None:
+        schema = get_schema_chart_config()
+    config_new = validate_chart_config_and_set_defaults(config_new, schema)
+    # Update map tab
+    config_new = update_chart_config_map(config_new, indicator_mapping)
+    # Update dimensions
+    config_new = update_chart_config_dimensions(config_new, indicator_mapping)
+    # Update sorting
+    config_new = update_chart_config_sort(config_new, indicator_mapping)
+    # Update indicator references
+    return config_new
+
+
+def update_chart_config_map(
```

Comment:
Not for this PR, but perhaps having `ChartConfig` object with convenient methods would be helpful in multiple places. (It'd also help with autocomplete)